### PR TITLE
Add Codemirror 6 to submission form

### PR DESCRIPTION
**Pull request by vidya-ram**

### CODE REVIEW

**Comment by miteshashar** [comment]
Submission form views when there are no labels created for the project.

*New Submission*
image

*Edit Submission*
image

The placement of the 3 possible fields here should be handled such that they are aesthetically placed in the UI.

1 field: Left
2 fields: Left, Right
3 fields: Left, Center, Right

**Comment by miteshashar** [comment]
There is no way to go back from the Add Collaborator popup. The only way is reloading the page, which would mean all unsaved changes are lost.

image

**Comment by vidya-ram** [comment]
> Submission form views when there are no labels created for the project.
> 
> **New Submission** image
> 
> **Edit Submission** image
> 
> The placement of the 3 possible fields here should be handled such that they are aesthetically placed in the UI.
> 
> 1 field: Left 2 fields: Left, Right 3 fields: Left, Center, Right

Fixed

**Comment by vidya-ram** [comment]
> There is no way to go back from the Add Collaborator popup, in case one does not want to add one. image The only way is reloading the page, which would mean all unsaved changes are lost.
> 
> Moreover, the user dropdown does not load any values. image
> 
> Clicking 'Add Collaborator' goes to the API url(losing all changes made to content). On the new page, the user field renders as an autocomplete field, though in the modal, it renders as a dropdown. image
> 
> Submitting this form successfully gets processed, but replaces the full page with JSON. image

Fixed

**Comment by miteshashar** [comment]
Submitting blank form shows a toastr message with text "parsererror".

image

**Comment by vidya-ram** [comment]
> Submitting blank form shows a toastr message with text "parsererror".
> 
> image

Fixed

File: funnel/static/js/scripts.js [hunk 1]
```diff
@@ -72,6 +72,14 @@ function activate_widgets() {
   });
 }
 
+function activate_select2() {
+  /* Upgrade to jquery 3.6 select2 autofocus isn't working. This is to fix that problem.
+    select2/select2#5993  */
+  $(document).on('select2:open', function () {
+    document.querySelector('.select2-search__field').focus();
```
Comment (jace) [inline funnel/static/js/scripts.js:79]: Will this work if the page has multiple select2 widgets?

File: funnel/views/proposal.py [hunk 2]
```diff
@@ -247,33 +247,44 @@ def edit(self) -> ReturnRenderWith:
...
-                'html': render_template(
-                    'collaborator_list.html.jinja2',
-                    collaborators=[
-                        _m.current_access(datasets=['primary', 'related'])
-                        for _m in self.obj.memberships
-                    ],
-                ),
-            }, 201
+        if request.method == 'POST':
+            if collaborator_form.validate_on_submit():
```
Comment (jace) [inline funnel/views/proposal.py:251]: `form.validate_on_submit()` is shorthand for `request.method == 'POST' and form.validate()` so the first `if` statement is not required.

**Review by jace — CHANGES_REQUESTED** [review]
One redundant `if` statement in the Python code.

**Pull request merged**